lib/puppet/catalog-diff/compilecatalog.rb

```diff
@@ -97,14 +97,14 @@ def compile_catalog(node_name, server, certless)
       end
 
       uri = URI("https://#{server}:#{port}#{endpoint}")
-      Puppet.debug("Connecting to server: #{uri}")
+      ssl_context = Puppet::CatalogDiff::Tlsfactory.ssl_context(tls_cert, tls_key, tls_ca)
       begin
         ret = if certless
-                Puppet.runtime[:http].post(uri, body.to_json, headers: headers)
+                Puppet.runtime[:http].post(uri, body.to_json, headers: headers, options: { ssl_context: ssl_context })
```

Comment:
doesn't this now make `certless` require certs (to be provided)?

Reply:
the name `certless` is a bit bogus here. if this boolean toggle is true, we talk to a puppetserver API with the name `certless`. This endpoint requires a TLS client cert to authenticate. Naming things is hard... 
It's this endpoint: https://puppet.com/docs/puppet/7/server/puppet-api/v4/catalog.html

It allows you to request a catalog for a common name that's not listed in the client cert you authenticate with.

Reply:
:bulb: 

so really `these-certs-are-made-for-requesting-everything`, not `certless`, gotcha

Reply:
yes :D